### PR TITLE
library: port Spinner demo to Vala

### PR DESCRIPTION
**Pull request by mar-v-in**

### CODE REVIEW

**Comment by sonnyp** [comment]
Thanks!

Please add your name to `about.js` in the contributors section :pray:

File: src/Library/demos/Spinner/main.vala [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env -S vala workbench.vala --pkg gtk4 --pkg libadwaita-1
+
+public void main () {
+  Gtk.init ();
+
+  var button = workbench.builder.get_object ("button") as Gtk.Button;
```
Comment (lw64) [inline src/Library/demos/Spinner/main.vala:6]: Please use here the (Gtk.Button) casting syntax.

**Review by lw64 — CHANGES_REQUESTED** [review]
Looks good overall. Thanks!

**Comment by sonnyp** [comment]
ping @mar-v-in :)

File: src/Library/demos/Spinner/main.vala (hunk 1 above)
Comment (mar-v-in) [reply src/Library/demos/Spinner/main.vala]: Changed.

I also checked other examples in the Library, and about half of them are using checked casts for the result of `builder.get_object()` whereas the other half is using unchecked casts. I guess that should be unified to not cause unnecessary confusion to beginners.

Comment (lw64) [reply src/Library/demos/Spinner/main.vala]: hm. seems like I didnt review them (properly). That should be done in a separate PR though.

**Review by lw64 — APPROVED** [review]

**Pull request merged**